core/primitives/src/version.rs

```diff
@@ -176,11 +176,15 @@ pub const PEER_MIN_ALLOWED_PROTOCOL_VERSION: ProtocolVersion = STABLE_PROTOCOL_V
 const STABLE_PROTOCOL_VERSION: ProtocolVersion = 55;
 
 /// Version used by this binary.
-#[cfg(not(feature = "nightly_protocol"))]
+#[cfg(all(not(feature = "nightly_protocol"), not(feature="shardnet")))]
 pub const PROTOCOL_VERSION: ProtocolVersion = STABLE_PROTOCOL_VERSION;
 /// Current latest nightly version of the protocol.
-#[cfg(feature = "nightly_protocol")]
+#[cfg(all(feature = "nightly_protocol", not(feature="shardnet")))]
```

Comment:
Please be consistent about formatting.

```suggestion
#[cfg(all(feature = "nightly_protocol", not(feature = "shardnet")))]
```

Reply:
done.